airbyte-connector-builder-server/unit_tests/connector_builder/impl/test_default_api.py

```diff
@@ -267,6 +271,85 @@ def test_read_stream_with_logs():
             assert actual_log == expected_logs[i]
 
 
+@pytest.mark.parametrize(
+    "request_record_limit, max_record_limit",
+    [
+        pytest.param(None, 3, id="test_create_request_no_record_limit"),
+        pytest.param(None, 1, id="test_create_request_no_record_limit_n_records_exceed_max"),
+        pytest.param(1, 3, id="test_create_request_with_record_limit"),
+        pytest.param(3, 1, id="test_create_request_record_limit_exceeds_max"),
+    ],
+)
+def test_read_stream_record_limit(request_record_limit, max_record_limit):
+    request = {
+        "url": "https://demonslayers.com/api/v1/hashiras?era=taisho",
+        "headers": {"Content-Type": "application/json"},
+        "body": {"custom": "field"},
+    }
+    response = {"status_code": 200, "headers": {"field": "value"}, "body": '{"name": "field"}'}
+    mock_source_adapter = MagicMock()
+    mock_source_adapter.read_stream.return_value = iter([
+        request_log_message(request),
+        response_log_message(response),
+        record_message("hashiras", {"name": "Shinobu Kocho"}),
+        record_message("hashiras", {"name": "Muichiro Tokito"}),
+        request_log_message(request),
+        response_log_message(response),
+        record_message("hashiras", {"name": "Mitsuri Kanroji"}),
+        response_log_message(response),
+    ])
+    n_records = 2
+    if request_record_limit is None:
+        record_limit = max_record_limit
+    else:
+        record_limit = min(request_record_limit, max_record_limit)
+
+    with patch.object(DefaultApiImpl, "MAX_RECORD_LIMIT", max_record_limit):
```

Comment:
feels like having to patch in this way breaks encapsulation. Does it make sense to make this an optional param on `DefaultApiImpl`?

Reply:
I agree with you, doesn't quite feel right to have `MAX_RECORD_LIMIT` on `DefaultApiImpl`. I'll move it to a constant outside of the class.

Reply:
I just meant it would be better to patch it using depenency injection i.e: make this an init variable for `DefaultApiImpl`

Reply:
Okay cool, I think that could be nice and tidy. Will make that change.